### ISSUE TITLE
Strange Issue Observed at AdFenix

### ISSUE THREAD

**Issue by atifwaqar** [Starcounter/Starcounter#4127]
SC Version : 2.3.0.5828

I just experienced a very strange issue, I am unable to reproduce it, but it happened twice for me. 
I have a page which contains a view-model with 2 nested child objects like this:
```
{
	"$CurrentPublicationOwnerDetails": {
		"Bind": "CurrentPublicationOwnerDetails_"
	},
	"CurrentPublicationOwnerDetails": {
		"parent": {
			"ObjectId": "",
			"child": {
				"ObjectId": "",
				"grandchild": {
					"ObjectId": ""
				}
			}
		}
	}
}
```
and code in JsonCodeBehind 
```
            public Database.PublicationOwner CurrentPublicationOwnerDetails_
            {
                get
                {
                    return Db.SQL<Database.PublicationOwner>("SELECT po FROM AdFenix.Database.PublicationOwner po WHERE ObjectId = ?", this.SelectedPublicationOwnerID).First;
                }
            }

```

The issue that I saw was that in my browser (chrome) data against `child` object was not correct, instead, it got a wrong child object with some other ObId.

**Comment by dan31** [comment]
CC @chrhol

**Comment by chrhol** [comment]
Yes sounds strange.

Is `parent` and `child` properties in this example bound as well?
And is this just a sample or a piece of existing code?

**Comment by atifwaqar** [comment]
> Is parent and child properties in this example bound as well?

If by bound you mean they have the same structure in DB then yes. JSON in my view-model is same as my Db Table.

> And is this just a sample or a piece of existing code?

This is a piece of existing code simplified in a parent-child example. actual code looks like this: 
```
  "$CurrentPublicationOwnerDetails": { "Bind": "CurrentPublicationOwnerDetails_" },
  "CurrentPublicationOwnerDetails": {
    "ObId": "",
    "ObjectID": "",
    "Advertiser": {
      "ObId": "",
      "FacebookPage": {
        "ObId": "",
      }
    }
  }
```
One thing I can see here is maybe `ObId` is messing with some objects cuz its case insensitivity in queries. But I never experienced this sort of issue in 2.2. (Or maybe I never noticed.)

**Comment by chrhol** [comment]
> One thing I can see here is maybe ObId is messing with some objects cuz its case insensitivity in queries. 

Might be, but if that was the case, then I think you would get an `CurrentPublicationOwnerDetails` object that you didn't expect and not an incorrect childobject. 

The incorrect childobject, is it belonging to some completely different parent (or maybe not belonging here at all), or can it be that the reference in the databaseobject is updated at the same time it's used in a viewmodel?

**Comment by atifwaqar** [comment]
> Might be, but if that was the case, then I think you would get an CurrentPublicationOwnerDetails object that you didn't expect and not an incorrect childobject.

I got the correct `CurrentPublicationOwnerDetails` with an incorrect `childobject`

> The incorrect childobject, is it belonging to some completely different parent (or maybe not belonging here at all)

Yes, its belonging to some completely different parent.


> or can it be that the reference in the databaseobject is updated at the same time it's used in a viewmodel?

No this is not the case.

**Comment by chrhol** [comment]
Alright, so something quite strange is going on then. As to what goes wrong is hard to say at this moment.

I see that the `Advertiser` property is also doing a query to get the reference, so (for now at least) this looks like a possible candidate for returning an unexpected object. 

If we're trying to rule out possible sources where this error can happen, I can't really see anything in Starcounter.XSON that would lead to this behaviour. The viewmodel will simply attach the object it gets from `AdFenix.Database.PublicationOwner.Advertiser`

**Comment by atifwaqar** [comment]
I agree with @chrhol . And this is why title of this issue sounds like a book title :P
Also, I tried to reproduce it again today, but it's not reproducing. I will surely add more details here as soon as I experience it again.
